raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/TraceEmitTasksPunctuator.java

```diff
@@ -70,24 +67,24 @@ class TraceEmitPunctuator implements Punctuator {
   private static final ConcurrentMap<String, Counter> tenantToTraceWithDuplicateSpansCounter =
       new ConcurrentHashMap<>();
 
+  private static final DropTask DROP_TASK = new DropTask();
   private final double dataflowSamplingPercent;
-  private final TraceIdentity key;
   private final ProcessorContext context;
   private final KeyValueStore<SpanIdentity, RawSpan> spanStore;
   private final KeyValueStore<TraceIdentity, TraceState> traceStateStore;
   private final To outputTopicProducer;
   private final long groupingWindowTimeoutMs;
-  private Cancellable cancellable;
 
-  TraceEmitPunctuator(
-      TraceIdentity key,
+  TraceEmitTasksPunctuator(
```

Comment:
We observed for some reason, the bytes read & written compaction rate has increased for span-data-store after this PR. Still looking into why it is happening, hopefully we can fix it

Reply:
> the bytes read & written compaction rate has increased 
False alarm. No increase in actual iops.